### DISCUSSION TITLE
Send OTP to the user on email but getting error for that API.

### DISCUSSION THREAD

**Discussion by snehalsabade** [Sunbird-Ed/Community#693] [General Discussion]
Hello all,
**Need Help**
We are customising registration flow with react. In user registration flow, we implemented google captcha. Next we are trying to sendOtp to the user on their email but getting api error 418I'm a Teapot.
We are trying to sendOtp to the user on email .
In google captcha we get successfully google captcha response.
Passing captcha response in headers.
getting response -   
**{
    "id": "api.validate.recaptcha",
    "ts": "2024-02-27T10:42:12.325Z",
    "params": {
        "resmsgid": "dd667150-d55c-11ee-b7a1-f3e82ff5a0c4",
        "msgid": "dd667151-d55c-11ee-b7a1-f3e82ff5a0c4",
        "err": "I'm a teapot",
        "status": "I'm a teapot",
        "errmsg": "I'm a teapot"
    },
    "responseCode": "I'm a teapot",
    "result": {}
}**
curl --location '{{host}}/learner/anonymous/otp/v1/generate?captchaResponse=03AFcWeA7iqBzBr7Huon9Im_JFahr-6a5Eq_Q6PZGMYzm4a7YVN_BqHPFfYrWEyojuL6gN4BLcheJXuGAWe08LN8YZGAde4kxQuw6goGqZctpx7jJlQaV3-Vpufu-g12ajs7h5BS5IbqRGxWP6dtrJNpqB0itxEmnJwrx4PxEXjJpq9noFXyTiPi-nZg3O_hLHZWUv0Ha6dnU7Nn5kh_T1mCViMk868j9A16kP1YBLfgE6vMH9K7rwsvqVLC62v1nGKPyHjMiz05uvqOELna-l2KgXhrlOzht3b1wovPHVxWBfA68xkem5oscuYK8WPpipLUWUUTvDGpsnRQfCN_WRBZDnnK8IUaCm4j7nnZhEp6Uhc25wlVNZ2p7Jm2RAIUi8n5VjPzxhwQDOW22ZIrb20gQsttQPu8FlzI027aMDrm-tfS48jVGm-KsaiwexDnB7AFVe56bByRLoflisTiRsggqXGQruWvwJtmdiFU1Du6uoXWFoViw30PCq2ubcWTDCYxeZMBLCsNA211nrZL0-rk1iGul_Q_jR6WFkSelGPG98dxOo8ijspWK5Qi4aq-V-QKWRUGaSzyXj4RMf-Z1WfJTPCXi0F_zruApIdlsVEW_IunGzAkifD56JB3UAya2qGjt82eqDdTpK3h_llYPb3rvbXbSb4tfwmwu-fOg-d5Bk6T44hTP4iWWKfireXXeMEip_Pu7fFhyIdaBR0eF2fcawdww2naK4MHKxo3_e_3W9HyEpW5Svc14Zcib2ZhqwIXQMwCxf8Cg7BRSJla60QJOiAZztHifAKBaAw_wOBA3AwZKNR82I9V2mkpj44J5vc1NoIwJd4On5djL2X6JLt07blhOh9rVDwyfV0V1Jv0xdK1JLHCov2bkvBAgwbusEW84-RIIGghtNm4n_0doty26K8hVj44DBzCdlMuEuEsTHB8wMF80x_B0OPYCG1bh4Q9uooyUi5n5jtLaaPwN9JbXOvedO2EOP8DkttWq6Uo1J2diC-hRh2F_czw9ff5YIZXw1hnd62A-xipSrF8M7-OjzNHLdArZ54TMs77DkenzK6g5hLLPgkSnwsOeEeeZH4fdLDGMkSEvq-_hYMZpt7OhnHHevKpewrR-vrwT4zMfrAafo5m4KivlGwZdQVy5TQiW27vh06yNe2QxzkwQL-YQLwmTcHg3h40Dl6ooJmjX-fWjA8cUsXfTAlxcSrDtpLJtmpihcXFVzkeRFXhAWQH8qqjfw-UNwFUN8VOD5G-YtPI_myTTjVWuHgVBuxFfcsW3HoQtGHbE55h6l09Riti0JQF1kwnbHPPHpE068BNnKEryFqrT4XS8jYZiPCmNoKysxpTQ-cUMwxfGQExZkIChXYt9_FGCMMKr9HVVewuT6GNFlu5cmd3MVKg2n0k7EElcdDlIDRrYSL7X9dOKZLOyfzo6emvQnAJEPURl_MYqs_UryA3dC2tDX2oxSaRdkFSeDHKrbuYt0nX5bjbuIZ6ti2uWxo20PkhQ7egsc5uo1Zd6-ib4hkBMtXzm6wHpRO619XqdPO-qR1FKW7lN0DJ3l6R2ac5uv77-27mdl43UVy3hOu-bC5nceRru2L_yazDkTIcXAUoqehLxXE4Vt4Ho4fFVerIIS-vis5KVPMxCePpghXzXxSfO3zGi5t9kFhz-cQI4GOU1EQo58l63FLUljhVxJIc8cD3G0Nn9mpcwq5EiPla1M5-g' \
--header 'Accept: application/json' \
--header 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--header 'Cookie: {{cookie}}' \
--header 'Origin: {{host}}' \
--header 'Referer: {{host}}/signup?client_id=portal&state=88ad52fd-0a51-457d-9b92-976877007223&redirect_uri=https:%2F%2F{{}}%2Fresources%3Fauth_callback%3D1&scope=openid&response_type=code&version=4&error_callback=https:%2F%2F{{}}%2Fauth%2Frealms%2Fsunbird%2Fprotocol%2Fopenid-connect%2Fauth' \
--header 'Sec-Fetch-Dest: empty' \
--header 'Sec-Fetch-Mode: cors' \
--header 'Sec-Fetch-Site: same-origin' \
--header 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36' \
--header 'X-App-Id: {{}}' \
--header 'X-App-Version: 4.10.0' \
--header 'X-Channel-Id: {{}}' \
--header 'X-Device-ID: 759396ed9ad716856e491533d6e1ffc6' \
--header 'X-Org-code: {{}}' \
--header 'X-Request-ID: 1f55a81a-287a-9653-f30d-2ac8332653e1' \
--header 'X-Session-ID: d3d87691-389b-0be5-fd09-cc0f775168ad' \
--header 'X-Source: web' \
--header 'X-msgid: 1f55a81a-287a-9653-f30d-2ac8332653e1' \
--header 'sec-ch-ua: "Not_A Brand";v="8", "Chromium";v="120", "Google Chrome";v="120"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'sec-ch-ua-platform: "Linux"' \
--header 'ts: 2024-02-27T14:52:44+05:30' \
--data-raw '{"request":{"key":"abc@yopmail.com","type":"email"}}'

cc: @paritshivani @komalm @manishakapadnis

**Comment by gandham-santhosh** [comment]
@snehalsabade check if the request is reaching to the right email server. Check sendEmail function is working independently 
@AmiableAnil please share your thoughts
